### PR TITLE
ADLS gen1-gen2-mapping

### PR DESCRIPTION
**Pull request by gapra-msft**

### CODE REVIEW

**Comment by alzimmermsft** [comment]
Looking at this mapping would we want to add an exists method into Datalake? It exists (no pun intended) in Blobs, Files, and Queues and was functionality in the previous version.

Were the create flags inverted in Datalake compared to Blobs? In Blobs it won't overwrite by default (unless that changed and I missed it) and it seems strange for two closely related SDKs to have such a difference.

Should there be an API mapping comment for the remove ACL functions explaining how setAcl is capable of accomplishing this functionality (if this work similarly to how blobs and queues handle it).

Were setExpiryTime and setTimes replaced with another API that encompasses their functionality?

**Comment by gapra-msft** [comment]
> Looking at this mapping would we want to add an exists method into Datalake? It exists (no pun intended) in Blobs, Files, and Queues and was functionality in the previous version.
> 
> Were the create flags inverted in Datalake compared to Blobs? In Blobs it won't overwrite by default (unless that changed and I missed it) and it seems strange for two closely related SDKs to have such a difference.
> 
> Should there be an API mapping comment for the remove ACL functions explaining how setAcl is capable of accomplishing this functionality (if this work similarly to how blobs and queues handle it).
> 
> Were setExpiryTime and setTimes replaced with another API that encompasses their functionality?
> 
> This is just general comments on the API and not for this PR.

Yep, you're right! I created an issue for the exists method https://github.com/Azure/azure-sdk-for-java/issues/6437 and the no overwrite by default is in my TODO list as well https://github.com/Azure/azure-sdk-for-java/issues/6104

File: sdk/storage/azure-storage-file-datalake/GEN1_GEN2_MAPPING.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+<td>checkAccess</td>
+<td>Checks whether the calling user has the required permissions for the file/directory.</td>
+<td>N/A</td>
+<td></td>
+</tr>
+<tr>
+<td>checkExists</td>
+<td>Check whether a file or directory exists.</td>
+<td>N/A</td>
+<td>Use Gen2 API methods <strong>createFileWithResponse</strong> and <strong>createDirectoryWithResponse</strong> and set <strong>BlobRequestConditions.setIfNoneMatch(true)</strong></td>
```
Comment (rickle-msft) [inline sdk/storage/azure-storage-file-datalake/GEN1_GEN2_MAPPING.md]: I think this needs to be setIfNoneMatch("*"). And didn't I also see you create an issue for adding an exists method()? That would be a better mapping

Comment (gapra-msft) [reply sdk/storage/azure-storage-file-datalake/GEN1_GEN2_MAPPING.md]: You're right, thanks for that catch. And I created the two issues for exists and no overwrite default, we'll have to update the doc once those features exist

File: sdk/storage/azure-storage-file-datalake/GEN1_GEN2_MAPPING.md [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+<tr>
+<td>concatenateFiles</td>
+<td>Concatenate the specified list of files into this file.</td>
+<td>N/A</td>
+<td></td>
+</tr>
+<tr>
+<td>createClient</td>
+<td>Gets an ADLStoreClient object.</td>
+<td><strong>DataLakeServiceClient.create</strong></td>
```
Comment (rickle-msft) [inline sdk/storage/azure-storage-file-datalake/GEN1_GEN2_MAPPING.md]: ServiceClientBuilder.build

File: sdk/storage/azure-storage-file-datalake/GEN1_GEN2_MAPPING.md [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+<td>createDirectory</td>
+<td>Creates a directory and all it's parent directories if they do not already exist.</td>
+<td><strong>DataLakeDirectoryClient.create</strong></td>
+<td></td>
+</tr>
+<tr>
+<td>createEmptyFile</td>
+<td>Creates an empty file.</td>
+<td><strong>DataLakeFileClient.create</strong></td>
+<td>By default this method will override an existing file, set <strong>BlobRequestConditions.setIfNoneMatch(true)</strong> to prevent overwriting.</td>
```
Comment (rickle-msft) [inline sdk/storage/azure-storage-file-datalake/GEN1_GEN2_MAPPING.md]: "will overwrite"; ifNoneMatch("*")

File: sdk/storage/azure-storage-file-datalake/GEN1_GEN2_MAPPING.md [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+<td>enumerateDirectory</td>
+<td>Enumerates the contents of a directory, returning a List of Directory Entry objects, one per file or directory in the specified directory.</td>
+<td><strong>DataLakeFileSystemClient.listPaths</strong></td>
+<td>Set <strong>ListPathsOptions.path</strong> to the desired path</td>
+</tr>
+<tr>
+<td>getAclStatus</td>
+<td>Queries the ACLs and permissions for a file or directory.</td>
+<td><strong>DataLakeFileClient.getAccessControl</strong> and <strong>DataLakeDirectoryClient.getAccessControl</strong></td>
+<td>Set <strong>ListPathsOptions.path</strong> to the desired path</td>
```
Comment (rickle-msft) [inline sdk/storage/azure-storage-file-datalake/GEN1_GEN2_MAPPING.md]: I don't think ListPathsOptions are relevant here

**Comment by gapra-msft** [comment]
Created an issue to provide enumerateDirectory functionality https://github.com/Azure/azure-sdk-for-java/issues/6442

File: sdk/storage/azure-storage-file-datalake/GEN1_GEN2_MAPPING.md (hunk 3 above)
Comment (gapra-msft) [reply sdk/storage/azure-storage-file-datalake/GEN1_GEN2_MAPPING.md]: fixed

File: sdk/storage/azure-storage-file-datalake/GEN1_GEN2_MAPPING.md (hunk 2 above)
Comment (gapra-msft) [reply sdk/storage/azure-storage-file-datalake/GEN1_GEN2_MAPPING.md]: fixed

File: sdk/storage/azure-storage-file-datalake/GEN1_GEN2_MAPPING.md (hunk 1 above)
Comment (gapra-msft) [reply sdk/storage/azure-storage-file-datalake/GEN1_GEN2_MAPPING.md]: fixed

File: sdk/storage/azure-storage-file-datalake/GEN1_GEN2_MAPPING.md (hunk 4 above)
Comment (gapra-msft) [reply sdk/storage/azure-storage-file-datalake/GEN1_GEN2_MAPPING.md]: fixed

**Pull request merged**